Source/WebCore/platform/text/CharacterProperties.h

```diff
@@ -98,4 +98,9 @@ inline bool isControlCharacter(UChar32 character)
     return u_charType(character) == U_CONTROL_CHAR;
 }
 
+inline bool isPrivateUseAreaCharacter(UChar32 character)
+{
+    return ublock_getCode(character) == UBlockCode::UBLOCK_PRIVATE_USE_AREA;
```

Comment:
Since there are multiple private use blocks, this would be better off doing `(UCharCategory)u_charType(character) == U_PRIVATE_USE_CHAR`.

Reply:
Thank you for the review!
I’m trying to understand what is the difference of a character whose block is `UBlockCode::UBLOCK_PRIVATE_USE_AREA` and a character whose general category is `U_PRIVATE_USE_CHAR`.

I thought the former would mean that the block which contains the unicode is one of: BMP, Plane15 or Plane16, which seems the same as a character that belongs to the general category "private use".

Reply:
A block is a range of characters, with `UBLOCK_PRIVATE_USE_AREA` being U+E000..FEFF. You raise a good issue, though, that it is possible for a platform to change the general category of private use characters (and I beleive the system version ICU does). I will double-check, but until then this function can also check for `UBLOCK_SUPPLEMENTARY_PRIVATE_USE_AREA_A` and `UBLOCK_SUPPLEMENTARY_PRIVATE_USE_AREA_B`.

Reply:
ah now I got it, `UBLOCK_PRIVATE_USE_AREA` is just accounting for one of the blocks indeed:

>      * Until Unicode 3.1.1, the corresponding block name was "Private Use",
>      * and multiple code point ranges had this block.
>      * Unicode 3.2 renames the block for the BMP PUA to "Private Use Area" and
>      * adds separate blocks for the supplementary PUAs.